src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -329,4 +336,137 @@ public Version since() {
 
 	}
 
+	@api
+	@noboilerplate
+	@seealso({x_new_thread.class})
+	public static class _synchronized extends AbstractFunction {
+		private static final HashMap<Object, Integer> syncObjectMap = new HashMap<Object, Integer>();
```

Comment:
HashMap VS TreeMap is O(1) VS O(log(n)) for get and put. I just went with a HashMap because there will rarely be more than 5 objects in there, I believe lookup speed won't be noticable anyways (the only way to compare which one is actually faster is by testing according to "google"). I will change the syncObjectMap to "Map".
The volatile keyword in java is for Object references that might be changed in between 2 calls by some other thread. In this case, the variable will never change (it's even declared final) so the volatile keyword does not apply (it's actually a compile error to use volatile with final).